lib/queue.rb

```diff
@@ -1,31 +1,94 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @size = 20
+    @front = -1
+    @back = -1
   end
-
-  def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+  
+  # add element to rear
+  def enqueue(value)
+    if @front == -1
+      @front = 0
+      @back = 1
+      @store[@front] = value
+    elsif @store.empty?
+      raise ArgumentError, "Full"
```

Comment:
If the Queue is empty raise an argument error that it's full?  

I think you want to check to see if it's full by doing `if (@back + 1) % @store.length == @front`